source/ProcessManager.Orchestrations.Tests/Unit/Processes/BRS_021/ForwardMeteredData/V1/Activities/FindReceiversActivity_Brs_021_ForwardMeteredData_V1Tests.cs

```diff
@@ -0,0 +1,408 @@
+
+        var meteringPointType = MeteringPointType.FromCode(meteringPointCode);
+        var startDate = InstantPattern.General.Parse("2025-01-01T00:00:00Z").Value;
+        var endDate = InstantPattern.General.Parse("2025-01-02T00:00:00Z").Value;
+        var childEnergySupplier = (ActorId: "6798000020100", ActorRole: ActorRole.EnergySupplier);
+
+        var electricityMarketChildMeteringPointMasterData =
+            SetupMeteringPointMasterData(
+                energySupplierId: childEnergySupplier.ActorId,
+                startDate: startDate,
+                endDate: endDate,
+                parentEnergySupplierId: expectedEnergySupplier.ActorId);
+
+        // Act
+        var result = await Sut.Run(
+            new FindReceiversActivity_Brs_021_ForwardMeteredData_V1.ActivityInput(
+                _orchestrationInstanceId,
+                meteringPointType.Code,
+                startDate.ToString(),
+                endDate.ToString(),
+                electricityMarketChildMeteringPointMasterData));
+
+        // Assert
+        result.MarketActorRecipients.Should().HaveCount(1);
+        result.MarketActorRecipients.Should()
+            .ContainSingle(
+                x => x.ActorId == expectedEnergySupplier.ActorId
+                     && x.ActorRole == expectedEnergySupplier.ActorRole);
+    }
+
+    [Fact]
+    public async Task FindReceiversActivity_WhenMeteringPointTypeIsAnalysisAndParentMeteringPointDoesNotExists_ReturnsNoReceiver()
+    {
+        // Arrange
+        var meteringPointType = MeteringPointType.Analysis;
+        var startDate = InstantPattern.General.Parse("2025-01-01T00:00:00Z").Value;
+        var endDate = InstantPattern.General.Parse("2025-01-02T00:00:00Z").Value;
+        var energySupplier = (ActorId: "6798000020100", ActorRole: ActorRole.EnergySupplier);
+
+        var electricityMarketChildMeteringPointMasterData =
+            SetupMeteringPointMasterData(
+                energySupplierId: energySupplier.ActorId,
+                startDate: startDate,
+                endDate: endDate);
+
+        // Act
+        var result = await Sut.Run(
+            new FindReceiversActivity_Brs_021_ForwardMeteredData_V1.ActivityInput(
+                _orchestrationInstanceId,
+                meteringPointType.Code,
+                startDate.ToString(),
+                endDate.ToString(),
+                electricityMarketChildMeteringPointMasterData));
+
+        // Assert
+        result.MarketActorRecipients.Should().BeEmpty();
+    }
+
+    [Theory]
+    [InlineData("D14")]
+    [InlineData("D15")]
+    [InlineData("D19")]
+    public async Task FindReceiversActivity_WhenMeteringPointTypeIsAUnsupportedChildMeteringPoint_ReturnsNoReceiver(
+        string meteringPointCode)
+    {
+        // Arrange
+        var expectedEnergySupplier = (ActorId: "5798000020000", ActorRole: ActorRole.EnergySupplier);
```

Comment:
```suggestion
        var notExpectedEnergySupplier = (ActorId: "5798000020000", ActorRole: ActorRole.EnergySupplier);
```
"result.MarketActorRecipients.Should().HaveCount(0)" ?